### PR TITLE
Refactoring join mailchimp sync data without where

### PR DESCRIPTION
**Pull request by roberto-ebizmarts**

### CODE REVIEW

File: app/code/community/Ebizmarts/MailChimp/Model/Api/Carts.php [hunk 1]
```diff
@@ -271,7 +265,7 @@ public function _getNewQuotes()
         // be sure that the quotes are already in mailchimp and not deleted
         $this->getEcommerceQuoteCollection()->addWhere(
             $newCarts, "m4m.mailchimp_sync_delta IS NULL", $this->getBatchLimitFromConfig()
-        );
+        );*/
```
Comment (SantiParagarino) [inline app/code/community/Ebizmarts/MailChimp/Model/Api/Carts.php]: Please remove commented code

File: app/code/community/Ebizmarts/MailChimp/Model/Api/ItemSynchronizer.php [hunk 2]
```diff
@@ -318,4 +318,46 @@ protected function getItemType()
     {
         return null;
     }
+
+    public function joinMailchimpSyncDataWithoutWhere($customerCollection, $mailchimpStoreId)
+    {
+//        $this->getResourceCollection()->joinMailchimpSyncDataWithoutWhere($customerCollection, $mailchimpStoreId);
```
Comment (SantiParagarino) [inline app/code/community/Ebizmarts/MailChimp/Model/Api/ItemSynchronizer.php]: Please remove comment

File: app/code/community/Ebizmarts/MailChimp/Model/Api/Orders.php [hunk 3]
```diff
@@ -84,20 +86,27 @@ protected function _getModifiedOrders()
...
+         $this->_ecommerceOrdersCollection->joinLeftEcommerceSyncData($modifiedOrders);
+         // be sure that the order are already in mailchimp and not deleted
+         $this->_ecommerceOrdersCollection->addWhere(
+             $modifiedOrders,
+             "m4m.mailchimp_sync_modified = 1",
+             $this->getBatchLimitFromConfig()
+         );*/
+
+        $modifiedOrders = $this->buildEcommerceCollectionToSync(
+            Ebizmarts_MailChimp_Model_Config::IS_ORDER,
```
Comment (SantiParagarino) [inline app/code/community/Ebizmarts/MailChimp/Model/Api/Orders.php]: Please remove comments

File: app/code/community/Ebizmarts/MailChimp/Model/Api/Orders.php [hunk 4]
```diff
@@ -196,10 +206,11 @@ protected function _getNewOrders()
             $newOrders,
             "m4m.mailchimp_sync_delta IS NULL",
             $this->getBatchLimitFromConfig()
-        );
+        );*/
```
Comment (SantiParagarino) [inline app/code/community/Ebizmarts/MailChimp/Model/Api/Orders.php]: Please remove comments

File: app/code/community/Ebizmarts/MailChimp/Model/Api/Orders.php [hunk 5]
```diff
@@ -1149,4 +1152,32 @@ public function createEcommerceOrdersCollection()
...
+
+    protected function addFilters(
+        Mage_Sales_Model_Resource_Order_Collection $collectionToSync,
+        $isNewItem = "new"
+    ){
+        //Mage::log("VVVVV addFilters", null, "orders.log", true);
+        $magentoStoreId = $this->getMagentoStoreId();
+        //Mage::log("==== addFieldToFilter magentoStoreId: $magentoStoreId", null, "orders.log", true);
+        $collectionToSync->addFieldToFilter('store_id', array('eq' => $magentoStoreId));
+        //Mage::log("==== addFieldToFilter Filters added", null, "orders.log", true);
```
Comment (SantiParagarino) [inline app/code/community/Ebizmarts/MailChimp/Model/Api/Orders.php]: Remove logs commented

**Review by SantiParagarino — CHANGES_REQUESTED** [review]

File: app/code/community/Ebizmarts/MailChimp/Model/Api/Carts.php (hunk 1 above)
Comment (roberto-ebizmarts) [reply app/code/community/Ebizmarts/MailChimp/Model/Api/Carts.php]: Solved.

File: app/code/community/Ebizmarts/MailChimp/Model/Api/ItemSynchronizer.php (hunk 2 above)
Comment (roberto-ebizmarts) [reply app/code/community/Ebizmarts/MailChimp/Model/Api/ItemSynchronizer.php]: Solved.

File: app/code/community/Ebizmarts/MailChimp/Model/Api/Orders.php (hunk 3 above)
Comment (roberto-ebizmarts) [reply app/code/community/Ebizmarts/MailChimp/Model/Api/Orders.php]: Solved.

File: app/code/community/Ebizmarts/MailChimp/Model/Api/Orders.php (hunk 4 above)
Comment (roberto-ebizmarts) [reply app/code/community/Ebizmarts/MailChimp/Model/Api/Orders.php]: Solved.

File: app/code/community/Ebizmarts/MailChimp/Model/Api/Orders.php (hunk 5 above)
Comment (roberto-ebizmarts) [reply app/code/community/Ebizmarts/MailChimp/Model/Api/Orders.php]: Solved.

**Review by SantiParagarino — APPROVED** [review]

**Pull request merged**